src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -236,6 +236,29 @@ private MessagePayloadDataConverter CreateErrorDataConverter(IErrorSerializerSet
             return new MessagePayloadDataConverter(errorSerializerSettingsFactory.CreateJsonSerializerSettings(), isDefault);
         }
 
+        private IDurabilityProviderFactory GetDurabilityProviderFactory(ILogger logger, IEnumerable<IDurabilityProviderFactory> orchestrationServiceFactories)
+        {
+            bool storageTypeIsConfigured = this.Options.StorageProvider.TryGetValue("type", out object storageType);
+            if (!storageTypeIsConfigured)
+            {
+                logger.LogInformation("Using the default storage provider: Azure Storage.");
+                return orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals("Azure Storage"));
+            }
+
+            if (orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals(storageType)) == null)
+            {
+                IList<string> factoryNames = orchestrationServiceFactories.Select(f => f.Name).ToList();
+                logger.LogWarning($"Storage provider type ({storageType}) was not found. " +
+                    $"Available storage providers: {string.Join(", ", factoryNames)}. " +
+                    $"Using the default storage provider: Azure Storage.");
+
+                return orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals("Azure Storage"));
+            }
+
+            logger.LogInformation($"Using the {storageType} storage provider.");
+            return orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals(storageType));

```

Comment:
It looks like you're doing `orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals(storageType))` multiple times. Let's reduce this to just once:
```csharp
// Lookup is case-insensitive
IDurabilityProviderFactory selectedFactory = orchestrationServiceFactories.FirstOrDefault(
    f => f.Name.Equals(storageType, StringComparison.OrdinalIgnoreCase));
if (selectedFactory == null)
{
    // ...
}

return selectedFactory;
```